### PR TITLE
SR-3 Add the Recruiter, Leader, Team and Question models

### PR DESCRIPTION
**Pull request by Mendrejk**
Based on the MVP ERD: https://informatyzacja.atlassian.net/browse/SR-3

### CODE REVIEW

**Review by DocentSzachista — DISMISSED** [review]

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Recruiter.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package pl.edu.pwr.samorzad.informatyzacja.recruitmentsystem.model
+
+class Recruiter(
+    val id: Int,
+    val email: String,
+    val teams: MutableList<Team>
```
Comment (deamacto) [inline backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Recruiter.kt:6]: Nie lepiej zrobić MutableSet?

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Team.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package pl.edu.pwr.samorzad.informatyzacja.recruitmentsystem.model
+
+data class Team(
+    val id: Int,
+    val name: String,
+    val recruiters: MutableList<Recruiter>
```
Comment (deamacto) [inline backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Team.kt:6]: Tutaj też MutableSet

**Review by deamacto — CHANGES_REQUESTED** [review]

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Recruiter.kt (hunk 1 above)
Comment (Mendrejk) [reply backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Recruiter.kt]: Updated

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Team.kt (hunk 2 above)
Comment (Mendrejk) [reply backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Team.kt]: Updated

**Comment by Mendrejk** [comment]
Used MutableSets for collections of Teams and Recruiters. Overriden equals so that Leader and Recruiter are compared only by email and Team is compared only by name.

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Leader.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            return true
+        }
+
+        if (javaClass != other?.javaClass) {
+            return false
+        }
+
+        other as Leader
+
+        if (email != other.email) {
```
Comment (deamacto) [inline backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Leader.kt:19]: Może zamiast tego zrobić return email == other.email?

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Recruiter.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+            return true
+        }
+
+        if (javaClass != other?.javaClass) {
+            return false
+        }
+
+        other as Recruiter
+
+        if (email != other.email) {
```
Comment (deamacto) [inline backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Recruiter.kt:19]: Tutaj też return email == other.email

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Team.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+            return true
+        }
+
+        if (javaClass != other?.javaClass) {
+            return false
+        }
+
+        other as Team
+
+        if (name != other.name) {
```
Comment (deamacto) [inline backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Team.kt:19]: return name == other.name

**Review by deamacto — CHANGES_REQUESTED** [review]

**Comment by Mendrejk** [comment]
Simplified return statements

**Review by deamacto — APPROVED** [review]

**Review by DocentSzachista — APPROVED** [review]

File: backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Leader.kt [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    override fun equals(other: Any?): Boolean {
+        if (this === other) {
+            return true
+        }
+
+        if (javaClass != other?.javaClass) {
+            return false
+        }
+
+        other as Leader
```
Comment (Bravo555) [inline backend/recruitment-system/src/main/kotlin/pl/edu/pwr/samorzad/informatyzacja/recruitmentsystem/model/Leader.kt:17]: All looks straightforward, but could you tell me what's the purpose of this expression? The docs say that it's just a cast, but what's happening with the result of this cast? It isn't assigned to a variable and isn't returned. Is it modified in place?

**Review by Bravo555 — APPROVED** [review]

**Pull request merged**